File: vuln_eval.py
"""
Trigger rules:
  - tg-py-eval-user-input   (eval with user input)
  - tg-py-exec-user-input   (exec with user input)
CWE-95: Improper Neutralization of Directives in Dynamically Evaluated Code
"""

# ── 🔴 eval() with user input ──────────────────────────────────────────────
user_expr = input("Enter a math expression: ")
result = eval(user_expr)  # ❌ RULE HIT: tg-py-eval-user-input
print(f"Result: {result}")

# ── 🔴 eval() with request data (Flask-style) ──────────────────────────────
def compute(data):
    formula = data.get("formula", "1+1")
    return eval(formula)  # ❌ RULE HIT: tg-py-eval-user-input

# ── 🔴 exec() with user input ──────────────────────────────────────────────
code_snippet = input("Enter Python code to run: ")
exec(code_snippet)  # ❌ RULE HIT: tg-py-exec-user-input

# ── 🔴 exec() with external data ───────────────────────────────────────────
def run_plugin(plugin_code):
    exec(plugin_code)  # ❌ RULE HIT: tg-py-exec-user-input

File: vuln_eval.js
/**
 * Trigger rules:
 *   - tg-js-eval-injection  (eval with dynamic input)
 * CWE-95: Code Injection
 */
const express = require('express');
const app = express();

// ── 🔴 eval() with query parameter ────────────────────────────────────────
app.get('/calc', (req, res) => {
  const expr = req.query.expression;
  const result = eval(expr); // ❌ RULE HIT: tg-js-eval-injection
  res.json({ result });
});

// ── 🔴 eval() with request body ───────────────────────────────────────────
app.post('/execute', (req, res) => {
  const code = req.body.code;
  const output = eval(code); // ❌ RULE HIT: tg-js-eval-injection
  res.json({ output });
});

module.exports = app;
